edx_exams/apps/api/v1/views.py

```diff
@@ -440,36 +444,50 @@ def get(self, request):
         """
         HTTP GET handler to fetch all exam attempt data
 
+        TODO: The endpoint should be refactored on deprecation of edx-proctoring.
+        Most of the complexity here has to do with the fact that we are supporting
+        slightly different API behaviors from that service.
+
         Parameters:
             None
 
         Returns:
             A Response object containing all `ExamAttempt` data.
         """
         user = request.user
-        latest_attempt = get_latest_attempt_for_user(user.id)
-        serialized_attempt = StudentAttemptSerializer(latest_attempt)
-
-        # if there is an active attempt in this service, return it
-        if latest_attempt is not None:
-            latest_attempt = check_if_exam_timed_out(latest_attempt)
-            serialized_attempt = StudentAttemptSerializer(latest_attempt)
-            if latest_attempt.status in (ExamAttemptStatus.started, ExamAttemptStatus.ready_to_submit):
+        exam_content_id = request.GET.get('content_id', None)
+        exam = get_exam_by_content_id(exam_content_id)
+
+        # if a specific exam is requested always return that exam
+        if exam is not None:
+            attempt = get_current_exam_attempt(user.id, exam.id)
+        else:
+            attempt = get_active_attempt_for_user(user.id)
+
+        serialized_attempt = StudentAttemptSerializer(attempt)
```

Comment:
Would it be more efficient for this to be below the if statement block below? I'm thinking we might otherwise end up serializing this attempt twice for no reason.

Reply:
good call, there's something odd about that logic here